lib/actions/logs.coffee

```diff
@@ -42,14 +30,18 @@ module.exports =
 	]
 	permission: 'user'
 	action: (params, options, done) ->
-		resin.logs.subscribe params.uuid, {
-			history: options.num or LOGS_HISTORY_COUNT
-			tail: options.tail
-		}, (error, message) ->
-			return done(error) if error?
-			if _.isArray(message)
-				_.each message, (line) ->
+		if options.tail?
+			resin.logs.history(params.uuid).then (lines) ->
+				for line in lines
```

Comment:
I'm thinking we could use Bluebird's `.each()` here to make it even more compact: https://github.com/petkaantonov/bluebird/blob/master/API.md#eachfunction-iterator---promise.